cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -58,138 +60,278 @@ def __init__(
                 in the form of the t in (-1)**t = exp(i pi t).
 
         Raises:
-            ValueError: If the given pauli string does not have eignevalues +1
-                or -1.
+            ValueError: If coefficient is not 1 or -1.
         """
-        if pauli_string.coefficient == -1:
-            pauli_string = -pauli_string
-            exponent_pos, exponent_neg = exponent_neg, exponent_pos
+        gate = PauliStringPhasorGate(
+            dps.DensePauliString(pauli_string.values(), coefficient=pauli_string.coefficient),
+            exponent_neg=exponent_neg,
+            exponent_pos=exponent_pos,
+        )
+        super().__init__(gate, tuple(pauli_string))
```

Comment:
Please replace `tuple(pauli_string)` with `pauli_string.qubits`. The former is very hard to read.